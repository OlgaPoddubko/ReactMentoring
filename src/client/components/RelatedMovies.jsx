// @flow
import * as React from 'react';
import Movie from './Movie';
import { connect } from 'react-redux';

type Props = {
  relatedMovies: [],
}

class RelatedMovies extends React.Component<Props> {
  static defaultProps = {
    relatedMovies: [],
  };

  render() {
    const relatedMovies = this.props.relatedMovies;
    return (
      <div className="items">
        { relatedMovies.map(elem =>
            <Movie
              key={elem.id}
              movie={elem}
            />
          )
        }
        <style jsx>{`
          .items {
            max-width: 1200px;
            margin: 0 auto;
            padding-bottom: 90px;
          }
        `}</style>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  relatedMovies: state.gallery.relatedMovies,
});

export default connect(mapStateToProps)(RelatedMovies);
