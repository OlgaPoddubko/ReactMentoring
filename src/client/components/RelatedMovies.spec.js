import React from 'react';
import { RelatedMovies } from './RelatedMovies';
import moviesArr from '../../../test/moviesArr';

describe('render RelatedMovies', () => {
  it('should render RelatedMovies', () => {
    const wrapper = shallow(<RelatedMovies relatedMovies = { moviesArr.data } />);
    expect(wrapper).toMatchSnapshot();
  });
});
