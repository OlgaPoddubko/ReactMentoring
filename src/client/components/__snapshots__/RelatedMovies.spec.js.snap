// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render RelatedMovies should render RelatedMovies 1`] = `
<div
  className="items"
>
  <Connect(Movie)
    key="58431"
    movie={
      Object {
        "budget": 28000000,
        "genres": Array [
          "Romance",
          "Horror",
          "Comedy",
          "Thriller",
        ],
        "id": 58431,
        "overview": "A zombie outbreak has fallen upon the land in this reimagining of Jane Austenâ€™s classic tale of the tangled relationships between lovers from different social classes in 19th century England. Feisty heroine Elizabeth Bennet (Lily James) is a master of martial arts and weaponry and the handsome Mr. Darcy (Sam Riley) is a fierce zombie killer, yet the epitome of upper class prejudice. As the zombie outbreak intensifies, they must swallow their pride and join forces on the blood-soaked battlefield in order to conquer the undead once and for all.",
        "poster_path": "https://image.tmdb.org/t/p/w500/jsJlOhVHiz1xXr36616EI8tJ4b2.jpg",
        "release_date": "2016-02-04",
        "revenue": 16374328,
        "runtime": 108,
        "tagline": "Bloody lovely.",
        "title": "Pride and Prejudice and Zombies",
        "vote_average": 5.5,
        "vote_count": 826,
      }
    }
  />
  <Connect(Movie)
    key="4348"
    movie={
      Object {
        "budget": 28000000,
        "genres": Array [
          "Drama",
          "Romance",
        ],
        "id": 4348,
        "overview": "Pride & Prejudice is a humorous story of love and life among English gentility during the Georgian era. Mr. Bennet is an English gentleman living in Hertfordshire with his overbearing wife and five daughters. If Mr. Bennet dies their house will be inherited by a distant cousin whom they have never met, so the family's future happiness and security is dependent on the daughters making good marriages.",
        "poster_path": "https://image.tmdb.org/t/p/w500/lAb9l4kgc6QWnHamBzTnskt71A7.jpg",
        "release_date": "2005-09-16",
        "revenue": 121147947,
        "runtime": 135,
        "tagline": "A romance ahead of its time.",
        "title": "Pride & Prejudice",
        "vote_average": 7.8,
        "vote_count": 1901,
      }
    }
  />
  <Connect(Movie)
    key="9732"
    movie={
      Object {
        "budget": 0,
        "genres": Array [
          "Adventure",
          "Animation",
          "Family",
        ],
        "id": 9732,
        "overview": "The circle of life continues for Simba, now fully grown and in his rightful place as the king of Pride Rock. Simba and Nala have given birth to a daughter, Kiara who's as rebellious as her father was. But Kiara drives her parents to distraction when she catches the eye of Kovu, the son of the evil lioness, Zira. Will Kovu steal Kiara's heart?",
        "poster_path": "https://image.tmdb.org/t/p/w500/9FHNYvpltLQfpakXpMMj7RsVHqm.jpg",
        "release_date": "1998-10-24",
        "revenue": 0,
        "runtime": 81,
        "tagline": "The Circle of Life Continues...",
        "title": "The Lion King 2: Simba's Pride",
        "vote_average": 6.7,
        "vote_count": 1412,
      }
    }
  />
  <style
    jsx={true}
  >
    
          .items {
            max-width: 1200px;
            margin: 0 auto;
            padding-bottom: 90px;
          }
        
  </style>
</div>
`;
